### PR TITLE
Add Firebase authentication to author task definition

### PR DESCRIPTION
**Pull request by ajmaddaford**
Add Firebase authentication to author task definition

### CODE REVIEW

**Comment by ajmaddaford** [comment]
[Integrate user sign in](https://trello.com/c/itrglzyg/296-integrate-user-sign-in)

File: author.tf [hunk 1]
```diff
@@ -44,12 +44,16 @@ data "template_file" "author" {
...
-    CONTAINER_TAG      = "${var.author_tag}"
-    AUTHOR_API         = "https://${aws_route53_record.author-api.fqdn}"
-    PUBLISHER_URL      = "https://${aws_route53_record.publisher.fqdn}"
+    LOG_GROUP           = "${aws_cloudwatch_log_group.author.name}"
+    CONTAINER_REGISTRY  = "${var.docker_registry}"
+    CONTAINER_TAG       = "${var.author_tag}"
+    AUTHOR_API          = "https://${aws_route53_record.author-api.fqdn}"
+    PUBLISHER_URL       = "https://${aws_route53_record.publisher.fqdn}"
     SURVEY_LAUNCHER_URL = "${var.survey_launcher_url}"
+    SKIP_AUTH           = "${var.skip_auth}"
```
Comment (jonnyshaw89) [inline author.tf]: Should we default this to false so we don't need to supply it?

**Comment by samiwel** [comment]
Looks good to me. Where should the variables get set, would this happen in the Concourse pipeline?

**Comment by ajmaddaford** [comment]
@samiwel For now staging will deploy as the developer Terraform does, without authentication. Jon is doing the work to integrate the smoke tests and as these haven't been updated to work with authentication they would fail. I've raised a pull request on https://github.com/ONSdigital/eq-deploy/pull/14 to add the firebase variables to the pre-prod deploy.

File: author.tf [hunk 2]
```diff
@@ -44,12 +44,16 @@ data "template_file" "author" {
...
+    LOG_GROUP           = "${aws_cloudwatch_log_group.author.name}"
+    CONTAINER_REGISTRY  = "${var.docker_registry}"
+    CONTAINER_TAG       = "${var.author_tag}"
+    AUTHOR_API          = "https://${aws_route53_record.author-api.fqdn}"
+    PUBLISHER_URL       = "https://${aws_route53_record.publisher.fqdn}"
     SURVEY_LAUNCHER_URL = "${var.survey_launcher_url}"
+    ENABLE_AUTH         = "${var.enable_auth}"
+    FIREBASE_PROJECT_ID = "${var.firebase_project_id}"
+    FIREBASE_API_KEY    = "${var.firebase_api_key}"
+    FIREBASE_MESSAGING_SENDER_ID = "${var.firebase_messaging_sender_id}"
```
Comment (jonnyshaw89) [inline author.tf]: You should fix the formatting here.

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

**Review by jonnyshaw89 — APPROVED** [review]

**Review by MIslam92 — APPROVED** [review]

**Comment by MIslam92** [comment]
@ajmaddaford "Squishy Squashy"

**Pull request merged**